### PR TITLE
Kapu's BetterLimbs port

### LINKED ISSUE #571: [BUG] Amputation head duplicate and warping.
**Round ID: [548](https://shiptest.ga/stats/rounds/548)**

## Testmerges:

Probably Kapulimbs.

## Reproduction:

Take a person, amputate their head. Amputate the head of another body. Place the head of the first amputee on the second, it will "reject" and send their head to the debug zone, replacing it with a bald caucasian head. Repeating the procedure with the "new" head on the same body will warp the new head to the body's position once the amputation surgery completes.

### LINKED ISSUE #612: Kapulimbs seem to not show sprites for some species[BUG]
## Testmerges:
The following pull requests are currently test merged:
#230: '[TEMP] Disable loadout costs' by AsciiSquid at commit a7399a3607
#459: 'Interrim Research Rework: Of Departments and Trading' by MemedHams at commit 5eb6e25a45
#522: 'Map zones and virtual levels' by Azarak at commit fdc1cd65ad
#505: 'Vinethud vote notification' by Zevotech at commit 7daed97ea6
#521: 'Xenomorph Tweaks' by MemedHams at commit 39935670fc
#530: 'Kapu's BetterLimbs port' by E-MonaRhg at commit eb416f1b2f
#572: 'The Dronetestening' by Pockets-byte at commit 82f0f99e42
#589: 'Spitfire 2: Geneva Conventions Be Damned' by swifterjack at commit 99b1bc9870

## Reproduction:
~~Get delimbed as, in my experience so far, a lizard, a kepori, a podperson, an ipc, a snail, a digitgrade lizard, or human. No limb sprite. Can't see it!~~

Maintainer edit:

Spawning a limb directly using `spawn` or a limbgrower will produce a limb with an invalid `icon` if it is `greyscale only`.

### LINKED ISSUE #5951: Flaky test planet_gen: atom_update_overlays overridden. Use override = TRUE to suppress this warning.
Flaky tests were detected in [this test run](https://github.com/shiptest-ss13/Shiptest/actions/runs/22558948243/attempts/1). This means that there was a failure that was cleared when the tests were simply restarted.
Failures:
```
planet_gen: [2026-03-02 02:31:26] Runtime in code/__HELPERS/unsorted.dm,1031: atom_update_overlays overridden. Use override = TRUE to suppress this warning.
Target: the polar bear (/mob/living/basic/bear/polar) Proc: add_fire_overlay
proc name: stack trace (/datum/proc/stack_trace)
src: /datum/element/perma_fire_over... (/datum/element/perma_fire_overlay)
call stack:
/datum/element/perma_fire_over... (/datum/element/perma_fire_overlay): stack trace("atom_update_overlays overridde...")
/datum/element/perma_fire_over... (/datum/element/perma_fire_overlay): RegisterSignal(the polar bear (/mob/living/basic/bear/polar), "atom_update_overlays", "add_fire_overlay", 0)
/datum/element/perma_fire_over... (/datum/element/perma_fire_overlay): Attach(the polar bear (/mob/living/basic/bear/polar))
the polar bear (/mob/living/basic/bear/polar):  AddElement(/list (/list))
the lava (8,20,5) (/turf/open/lava/smooth): burn stuff(null, 2)
the lava (8,20,5) (/turf/open/lava/smooth): process(2)
Objects (/datum/controller/subsystem/processing/obj): fire(0)
Objects (/datum/controller/subsystem/processing/obj): ignite(0)
Master (/datum/controller/master): RunQueue()
Master (/datum/controller/master): Loop(2)
Master (/datum/controller/master): StartProcessing(0) at code/__HELPERS/unsorted.dm:1031
```

### PR DESCRIPTION
**Pull request by ElorgRHG**
~~_I am 100% qualified to handle everything in this PR, also a good, small PR for a first time contributor, yes_~~
Will be subject to a LOTTA reviews, as currently its almost a 1:1 port.

## About The Pull Request
So basically this PR: https://github.com/BeeStation/BeeStation-Hornet/pull/5611, but on shiptest, guess I'll just copy-paste the original PR's description.

MAINTAINER EDIT:
Closes #612
Closes #571

Refactors limbs to retain their information as owners change. Also, makes most unique limbs their own type.
Improves digitigrade code significantly.

Ports digitigrade jumpsuits from ~~Skyrat~~ Oracle
OracleStation/OracleStation#920

## Why It's Good For The Game

Removes bad snowflake code, opens up bodypart modding.

_Also apparently allows Vox, yay_

## Changelog
:cl:Kapu1178, Quantum_m, Ziiro, E-MonaRhg, Marksuckerburg
add: Digitigrade legs now support some jumpsuits (more to be added later)
add: Digitigrade legs now support some hardsuits.
add: Almost every unique bodypart sprite is now a whole limb.
add: Limbgrower limbs are now a sickly green
refactor: Digitigrade legs are now full limbs.
refactor: Limbs retain information as they change owners
refactor: Bodypart code is generally less ass
code: Support for bodytype-based clothing.
fix: Lizard snoots no longer poke through hardsuit helmets
fix: CHANGELING TRANSFORM FUNCTIONS PROPERLY
del: IPC and Super Soldier mutation toxin.
code: Krokodil victims are now a subtype of human
/:cl:

~~CL is also 1:1~~

### CODE REVIEW

File: code/modules/surgery/bodyparts/dismemberment.dm [hunk 1]
```diff
@@ -18,7 +18,13 @@
 	var/obj/item/bodypart/affecting = C.get_bodypart(BODY_ZONE_CHEST)
 	affecting.receive_damage(clamp(brute_dam/2 * affecting.body_damage_coeff, 15, 50), clamp(burn_dam/2 * affecting.body_damage_coeff, 0, 50)) //Damage the chest based on limb's existing damage
 	C.visible_message("<span class='danger'><B>[C]'s [src.name] is violently dismembered!</B></span>")
-	INVOKE_ASYNC(C, /mob.proc/emote, "scream")
+
+	if(C.stat <= SOFT_CRIT)//No more screaming while unconsious
+		if(IS_ORGANIC_LIMB(affecting))//Chest is a good indicator for if a carbon is robotic in nature or not.
+			C.emote("scream")
```
Comment (Kapu1178) [inline code/modules/surgery/bodyparts/dismemberment.dm]:
```suggestion
			INVOKE_ASYNC(C, /mob.proc/emote, "scream")
```

**Comment by Kapu1178** [comment]
Some general notes for maintainers/the author:
1. Kepori/Teshari and Vox will need their own BODYTYPE_KEPORI and BODYTYPE_VOX flags, as they are not humanoid. Make sure to override their bodyparts' `bodytype` var with `BODYTYPE_KEPORI | BODYTYPE_ORGANIC`

2. I removed Synths & Military Synths on bee because their code was shit and nobody used them. They are almost 100% broken with this.

3. I left... alot... of unused vars in here. I need to clean them up. Also, `#define NUMBER_OF_BODYTYPES` is unused and can be removed.

**Comment by Dimasw99** [comment]
Dat removal of IPC mutation toxin tho. IPC shall be forever doomed now if they somehow got gibbed

**Comment by zestybastard** [comment]
> o

Doesn't RnD let you print IPC parts, or only organs?

**Comment by Ms-Mee** [comment]
> > o
> 
> Doesn't RnD let you print IPC parts, or only organs?

Organs only, yes.

**Comment by MarkSuckerberg** [comment]
FINALLY PASSES CHECKS:tm:

**Comment by ElorgRHG** [comment]
1. ~~Kepori/Teshari and Vox will need their own BODYTYPE_KEPORI and BODYTYPE_VOX flags, as they are not humanoid. Make sure to override their bodyparts' bodytype var with BODYTYPE_KEPORI | BODYTYPE_ORGANIC~~ Done, and there are no Vox... yet.

2. ~~I removed Synths & Military Synths on bee because their code was shit and nobody used them. They are almost 100% broken with this.~~ Also removed

3. ~~I left... alot... of unused vars in here. I need to clean them up. Also, #define NUMBER_OF_BODYTYPES is unused and can be removed.~~ Cleaned up

**Comment by tiramisuapimancer** [comment]
Why does this remove lizard, IPC, and rachnid ability to wear underwear !

File: code/__DEFINES/mobs.dm [hunk 2]
```diff
@@ -71,6 +71,45 @@
 #define LARVA_BODYPART "larva"
 #define DEVIL_BODYPART "devil"
 
+//Bodypart change blocking flags
+#define BP_BLOCK_CHANGE_SPECIES	(1<<0)
```
Comment (ZephyrTFA) [inline code/__DEFINES/mobs.dm]:
```suggestion
#define BP_BLOCK_CHANGE_SPECIES (1<<0)
```

Comment (Kapu1178) [reply code/__DEFINES/mobs.dm]: I HATE this lint.

**Comment by Kapu1178** [comment]
YOLO

**Pull request merged**